discovery_engine_core/core/src/document.rs

```diff
@@ -225,16 +225,6 @@ pub(crate) fn document_from_article(
     })
 }
 
-/// Discards documents where the title is duplicated, keeping only the top ranked.
-pub(crate) fn dedup_documents(documents: &mut Vec<Document>) {
-    documents.sort_unstable_by(|doc1, doc2| {
```

Comment:
This was used to de-dup documents when merging them in the engine. Since we no longer need that, and the `DuplicateFilter` provides the same functionality, I simply removed it.